tests/b2share_functional_tests/test_submission.py

```diff
@@ -47,122 +48,227 @@
+
+    bucket_links = get_bucket_links(record_create_res.headers)
+    assert len(bucket_links) == 1
+
+    record_url = url_for('invenio_records_rest.recuuid_item',
+                         pid_value=record_create_data['id'])
+
+    _test_get_record(client, record_url, record_create_data, bucket_links)
+
+    return (record_url, record_create_data, bucket_links)
+
+
+def get_bucket_links(headers):
+    header_links = headers.get_all('Link')
+    is_bucket = http_header_link_regex(RECORD_BUCKET_RELATION_TYPE)
+    bucket_links_full = [hl for hl in header_links if is_bucket.search(hl)]
+    return [link.split(';', 1)[0] for link in bucket_links_full]
+
+
+def _test_get_record(client, record_url, record_expected_data, bucket_links):
+    record_get_res = client.get(record_url, headers=json_headers)
+    assert record_get_res.status_code == 200
+    record_get_data = json.loads(
+        record_get_res.get_data(as_text=True))
+    assert '_internal' not in record_get_data['metadata']
+    assert record_get_data == record_expected_data
+    assert bucket_links == get_bucket_links(record_get_res.headers)
+
+
+def _test_patch_and_get_record(client, record_url, record_create_data, bucket_links):
+    patch = make_record_patch(record_create_data['metadata']['title'])
+    record_patch_res = client.patch(record_url,
+                                    data=json.dumps(patch),
+                                    headers=jsonpatch_headers)
+    assert record_patch_res.status_code == 200
+    record_patch_data = json.loads(record_patch_res.get_data(as_text=True))
+    assert '_internal' not in record_patch_data['metadata']
+
+    from jsonpatch import apply_patch
+    expected_data = apply_patch(record_create_data['metadata'], patch)
+    assert record_patch_data['metadata'] == expected_data
+    assert get_bucket_links(record_patch_res.headers) == bucket_links
+    _test_get_record(client, record_url, record_patch_data, bucket_links)
+
+
+def _test_put_and_get_record(client, record_url, record_create_data, bucket_links):
+    record_new_data = change_record(record_create_data['metadata'])
+    record_put_res = client.put(record_url,
+                                data=json.dumps(record_new_data),
+                                headers=json_headers)
+    assert record_put_res.status_code == 200
+    record_put_data = json.loads(record_put_res.get_data(as_text=True))
+    assert '_internal' not in record_put_data['metadata']
+    assert record_put_data['metadata'] == record_new_data
+    assert get_bucket_links(record_put_res.headers) == bucket_links
+    _test_get_record(client, record_url, record_put_data, bucket_links)
+
+
+def _test_file_ops(client, bucket_links, count):
+    for num in range(count):
+        # Test file upload
+        headers = {'Accept': '*/*'}
+        object_url = '{0}/{1}'.format(bucket_links[0], 'myfile.dat')
+        content = 'contents2~`!+üøåé+%20+ +\n{}'.format('word ' * num)
```

Comment:
Sorry if I missed something but I don't see what the reason for the cryptic string.